platform/sync-installer/sync-installer-impl/src/main/java/org/codice/ddf/sync/installer/impl/SynchronisedInstallerImpl.java

```diff
@@ -0,0 +1,566 @@
+
+  private BundleService bundleService;
+
+  public SynchronisedInstallerImpl(
+      BundleContext bundleContext,
+      ConfigurationAdmin configAdmin,
+      FeaturesService featuresService,
+      BundleService bundleService) {
+    this.bundleContext = bundleContext;
+    this.configAdmin = configAdmin;
+    this.featuresService = featuresService;
+    this.bundleService = bundleService;
+  }
+
+  @Override
+  public void waitForBootFinish() throws SynchronisedInstallerException {
+    waitForBootFinish(DEFAULT_MAX_FEATURE_WAIT);
+  }
+
+  @Override
+  public void waitForBootFinish(long maxWaitTime) throws SynchronisedInstallerException {
+    Callable<Boolean> isBootFinishedAvailable =
+        () -> {
+          LOGGER.debug("Waiting for BootFinished service to appear.");
+          return bundleContext.getServiceReference(BootFinished.class) != null;
+        };
+
+    wait(
+        isBootFinishedAvailable,
+        maxWaitTime,
+        DEFAULT_POLLING_INTERVAL,
+        String.format("BootFinished service did not appear within [%s] ms", maxWaitTime));
+    waitForBundles();
+  }
+
+  @Override
+  public Configuration createManagedFactoryService(
+      String factoryPid, Map<String, Object> properties, String bundleLocation)
+      throws SynchronisedInstallerException {
+    return createManagedFactoryService(
+        DEFAULT_MAX_SERVICE_WAIT, factoryPid, properties, bundleLocation);
+  }
+
+  @Override
+  public Configuration createManagedFactoryService(
+      long maxWaitTime, String factoryPid, Map<String, Object> properties, String bundleLocation)
+      throws SynchronisedInstallerException {
+    LOGGER.info("Creating managed service of factorypid [{}]", factoryPid);
+
+    Configuration createdConfig;
+    try {
+      createdConfig = configAdmin.createFactoryConfiguration(factoryPid, null);
+      createdConfig.setBundleLocation(bundleLocation);
+    } catch (IOException e) {
+      throw new SynchronisedInstallerException(
+          "Failed to initialize managed service factory configuration with fpid of [" + factoryPid + "]",
+          e);
+    }
+
+    try {
+      if (properties == null || properties.isEmpty()) {
+        createdConfig.update();
+      } else {
+        createdConfig.update(toDic(properties));
+      }
+    } catch (IOException e) {
+      throw new SynchronisedInstallerException(
+          "Failed to update created managed service factory configuration with pid of ["
+              + createdConfig.getPid()
+              + "]",
+          e);
+    }
+
+    waitForServiceAvailability(maxWaitTime, createdConfig.getPid());
+    return createdConfig;
+  }
+
+  @Override
+  public void updateManagedService(
+      String servicePid, Map<String, Object> properties, String bundleLocation)
+      throws SynchronisedInstallerException {
+    updateManagedService(DEFAULT_MAX_SERVICE_WAIT, servicePid, properties, bundleLocation);
+  }
+
+  @Override
+  public void updateManagedService(
```

Comment:
✏️ Do we realize that such an update will never catch all possible updates to properties of a given managed service? A lot of code goes directly to config admin including DDF's own config admin, felix config installer, ...